src/main/java/org/linksharing/server/auth/AuthController.java

```diff
@@ -43,8 +43,8 @@ public String registerUser(@Valid User user, BindingResult result) {
         String pass = user.getPassword();
         user.setPassword(passwordEncoder.encode(pass));
 
-        if (userRepository.existsByEmail(user.getEmail())) {  //doesn't work for some reason
-            return "redirect:/login?userExists";
+        if (userRepository.existsByEmail(user.getEmail())) {
+            return "redirect:/login?query=userExists";
```

Comment:
You already have a way to use http request params (see `param.logout` and `param.error`).
It's better to use the same approach across the project (naming convention).
https://dev.to/galwaycoder/why-naming-conventions-are-important-and-useful-in-software-development-d5j
Please, revert this liine to `return "redirect:/login?userExists";` and make use of it.

Reply:
Ok. Done.
Didn't quite understand how that works, now I think I do.